web/src/components/onboarding/taxId/OnboardingSingleTaxId.tsx

```diff
@@ -24,12 +22,8 @@ export const OnboardingSingleTaxId = ({
 
   const isTabletAndUp = useMediaQuery(MediaQueries.tabletAndUp);
   const { state, setProfileData } = useContext(ProfileDataContext);
-  const { isAuthenticated, setShowNeedsAccountModal } = useContext(NeedsAccountContext);
 
   const handleChange = (value: string): void => {
-    if (isAuthenticated === IsAuthenticated.FALSE) {
-      setShowNeedsAccountModal(true);
-    }
     if (handleChangeOverride) {
```

Comment:
So for this ticket I had a choice, 

This component is called in 2 places, in the tax task via taxInput via OnboardingTaxId and in the profile directly via OnboardingTaxId. 

Profile has this function declared in itself and passes this logic down as handleChangeOverride while the taxInput chose instead to use the logic above. 

I chose to follow the lead of profile and moved this logic above to the TaxInput and passed it down. 

I did this because it felt move discoverable and its a pattern we have all over profile to pass down the override instead of having it in the components itself (it also feels cleaner to have the component be more isolated) Also we needed some of this logic in the TaxInput anyways to prevent the save and register button from triggering as well.